react-ui/src/common/useVotingStatus.jsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * votingDisabled: currently mutating vote
+ * votingExpired: outside voting window
+ * votingUnavailable: votingDisabled || votingExpired
+ */
+
+import differenceInMilliseconds from 'date-fns/differenceInMilliseconds';
+import isFuture from 'date-fns/isFuture';
+import { useEffect, useState } from 'react';
+
+import useSwrContest from '../utils/useSwrContest';
+
+import useComponentsState from './useComponentsState';
+
+const MAX_DELAY = 2147483647;
```

Comment:
It's the maximum allowable value for `setTimeout`

Reply:
Ah, TIL. Does it really make sense to use this?

In the unlikely event that we have a voting window open longer than 24.8 days, and the user actually keeps their browser running and tab focused for that duration, to then fire the end voting code?

Perhaps we need to branch if the delay is longer than 24.8 days and reevaluate this condition after the timeout rather than firing the mutation and marking as expired.

On the other hand, this would be handling quite the edge case that is extremely unlikely to ever need to be accounted for, so maybe just add a comment that this is the max allowable value and call it a day?

Reply:
So I originally implemented this because the `demo` contest delay is longer than 24.8 days, and this caused the timeout to fire immediately. I went with the comment approach because as you pointed out, seems better to handle the edge case than restrict everything else to avoid triggering an edge case.